### PR TITLE
VideoCore: Implement Cached Writes, use fastmem for reading GPU memory and eliminate old stuffs

### PR DESCRIPTION
**Pull request by FernandoS27**
This PR reduces time spent reading data from CPU to GPU, reduces invalidation overhead, improves performance overall and fixes Pokemon Scarlet / Violet on Normal.

### CODE REVIEW

File: src/video_core/invalidation_accumulator.h [hunk 1]
```diff
@@ -0,0 +1,78 @@
+// SPDX-FileCopyrightText: Copyright 2018 yuzu Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <vector>
```
Comment (lioncash) [inline src/video_core/invalidation_accumulator.h:6]: `<utility>` also needs to be included for `std::pair`

File: src/video_core/invalidation_accumulator.h [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        buffer.emplace_back(start_address, accumulated_size);
+        for (auto& [address, size] : buffer) {
+            func(address, size);
+        }
+    }
+
+private:
+    static constexpr size_t atomicy_bits = 5;
+    static constexpr size_t atomicy_size = 1ULL << atomicy_bits;
+    static constexpr size_t atomicy_side_mask = atomicy_size - 1;
```
Comment (liamwhite) [inline src/video_core/invalidation_accumulator.h:70]: atomicity_size_mask

File: src/video_core/invalidation_accumulator.h [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            return;
+        }
+        buffer.emplace_back(start_address, accumulated_size);
+        for (auto& [address, size] : buffer) {
+            func(address, size);
+        }
+    }
+
+private:
+    static constexpr size_t atomicy_bits = 5;
```
Comment (liamwhite) [inline src/video_core/invalidation_accumulator.h:68]: `atomicity_`, not `atomicy_`

**Review by liamwhite — COMMENTED** [review]
Fix the test errors

File: src/video_core/memory_manager.cpp [hunk 4]
```diff
@@ -43,6 +46,11 @@ MemoryManager::MemoryManager(Core::System& system_, u64 address_space_bits_, u64
     big_page_table_cpu.resize(big_page_table_size);
     big_page_continous.resize(big_page_table_size / continous_bits, 0);
     entries.resize(page_table_size / 32, 0);
+    if (!Settings::IsGPULevelExtreme()) {
```
Comment (liamwhite) [inline src/video_core/memory_manager.cpp:49]: you must also disable if `!Settings::IsFastmemEnabled()`

Comment (FernandoS27) [reply src/video_core/memory_manager.cpp:49]: it will be null automatically though

Comment (liamwhite) [reply src/video_core/memory_manager.cpp:49]: no, it will not. if you do not do this it crashes immediately on boot

**Pull request merged**